### PR TITLE
chore(Sidebar): use EventStack component

### PR DESCRIPTION
**Pull request by layershifter**
We have a cool API called EventStack. It's time to make it more awesome 🐱  This PR offers a new component exposes the EventStack API as public and provides a declarative way to manage it.

This will allow to reduce complexility with subscribtions to events and will allow our users to work with our API deeply.

### CODE REVIEW

File: src/modules/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -191,6 +162,7 @@ class Sidebar extends Component {
       <Ref innerRef={this.handleRef}>
         <ElementType {...rest} className={classes}>
           {childrenUtils.isNil(children) ? content : children}
+          {visible && <EventStack name={'click'} on={this.handleDocumentClick} />}
```
Comment (layershifter) [inline src/modules/Sidebar/Sidebar.js]: 30 lines vs 1 line, exciting result ✌️

Comment (levithomason) [reply src/modules/Sidebar/Sidebar.js]: Very cool, I'll give this a proper review before merging.  It looks like a fantastic idea :D

**Comment by levithomason** [comment]
I assume the future of this component will change now that we're publishing EventStack from elsewhere (eventually Stardust)?

**Comment by layershifter** [comment]
Yep, I will move this component to a package.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.83.0`.

**Pull request merged**